src/common/io.py

```diff
@@ -1505,10 +1506,9 @@ def _get_interpolated_trajectory(self, data_index: int, start_index: int = 0, st
         """ Returns an interpolated trajectory for variable of corresponding index 'data_index'. """
         self._verify_file_data()
 
-        index_in_cache = data_index in self._data_2
-        partial_cache_ok = self._can_use_partial_cache(start_index, stop_index)
-        if (index_in_cache and not self._allow_file_updates) or (index_in_cache and partial_cache_ok):
-            return self._data_2[data_index]
+        current_data = self._data_2.get(data_index, None)
+        if (current_data is not None) and self._can_use_partial_cache(current_data, stop_index):
+            return self._data_2[data_index][start_index:stop_index]
```

Comment:
The slicing is new here, but looks correct to me. (Unless we fail to test some scenario)